util/data/packed_rrset.c

```diff
@@ -353,11 +354,12 @@ packed_rrset_copy_region(struct ub_packed_rrset_key* key,
 	for(i=0; i<d->count + d->rrsig_count; i++) {
 		if(d->rr_ttl[i] < now)
 			d->rr_ttl[i] = SERVE_EXPIRED?SERVE_EXPIRED_REPLY_TTL:0;
-		else	d->rr_ttl[i] -= now;
+		else	d->rr_ttl[i] -= SERVE_ORIGINAL_TTL ? data->ttl_add : now;
 	}
 	if(d->ttl < now)
 		d->ttl = SERVE_EXPIRED?SERVE_EXPIRED_REPLY_TTL:0;
-	else	d->ttl -= now;
+	else	d->ttl -= SERVE_ORIGINAL_TTL ? data->ttl_add : now;
+	d->ttl_add = 0; /* TTLs have been made relative */
```

Comment:
For my understanding, is this here "just to be safe", or really needed? Guess it is fine to have it either way.

Reply:
Since I don't know what will happen to the data in `struct packed_rrset_data` I added this just to be safe.